### PR TITLE
Jenkins 60935 Code churn metric

### PR DESCRIPTION
**Pull request by rathaya**

### CODE REVIEW

**Comment by uhafner** [comment]
I get a test error locally:
```
[ERROR] Failures: 
[ERROR]   GitMinerStepITest.shouldMineRepositoryIncrementally:94->verifyStatistics:121 
Expecting numberOfAuthors of:
  <io.jenkins.plugins.forensics.miner.FileStatistics@4f464ffe[authors={foo@jenkins.io},churn=2,creationTime=1600120167,fileName=source.txt,lastModificationTime=1600120166,linesOfCode=1,numberOfAddedLines={0db09e25ff4c95b228de19fe7bd735d78ee81e95=1, 13db24f0b2d1803b3addc81bc42e6e98ac1e75d3=1},numberOfAuthors=1,numberOfCommits=2,numberOfDeletedLines={0db09e25ff4c95b228de19fe7bd735d78ee81e95=0, 13db24f0b2d1803b3addc81bc42e6e98ac1e75d3=1}]>
to be:
  <2>
but was:
  <1>
[INFO] 
[ERROR] Tests run: 27, Failures: 1, Errors: 0, Skipped: 0

```

**Comment by rathaya** [comment]
> I get a test error locally:
> 
> ```
> [ERROR] Failures: 
> [ERROR]   GitMinerStepITest.shouldMineRepositoryIncrementally:94->verifyStatistics:121 
> Expecting numberOfAuthors of:
>   <io.jenkins.plugins.forensics.miner.FileStatistics@4f464ffe[authors={foo@jenkins.io},churn=2,creationTime=1600120167,fileName=source.txt,lastModificationTime=1600120166,linesOfCode=1,numberOfAddedLines={0db09e25ff4c95b228de19fe7bd735d78ee81e95=1, 13db24f0b2d1803b3addc81bc42e6e98ac1e75d3=1},numberOfAuthors=1,numberOfCommits=2,numberOfDeletedLines={0db09e25ff4c95b228de19fe7bd735d78ee81e95=0, 13db24f0b2d1803b3addc81bc42e6e98ac1e75d3=1}]>
> to be:
>   <2>
> but was:
>   <1>
> [INFO] 
> [ERROR] Tests run: 27, Failures: 1, Errors: 0, Skipped: 0
> ```

I noticed this today too. I am still trying to figure out why this happens. If I only run this one test, it works. If I run the whole test-class, it fails.

File: plugin/src/main/java/io/jenkins/plugins/forensics/git/miner/GitRepositoryMiner.java [hunk 1]
```diff
@@ -128,52 +144,81 @@ private RepositoryStatistics createStatisticsFromHead(final Repository repositor
...
+        private Map<String, LocChanges> getFilesAndDiffEntriesFromCommit(
+                final Repository repository,
+                final Git git, final String oldCommit,
                 final String newCommit, final FilteredLog logger) {
-            List<String> filePaths = new ArrayList<>();
-
-            try {
+            Map<String, LocChanges> filePaths = new HashMap<>();
+            OutputStream outputStream = DisabledOutputStream.INSTANCE;
+            ArrayList<FileHeader> fileHeaders = new ArrayList<>();
```
Comment (uhafner) [inline plugin/src/main/java/io/jenkins/plugins/forensics/git/miner/GitRepositoryMiner.java:189]:
```suggestion
            List<FileHeader> fileHeaders = new ArrayList<>();
```

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
I created a followup PR where I started to fix some minor problems. One of the tests is now failing on Windows:
https://ci.jenkins.io/job/Plugins/job/git-forensics-plugin/job/PR-173/2/testReport/io.jenkins.plugins.forensics.git.miner/GitMinerStepITest/windows_8___Build__windows_8____shouldMineRepositoryIncrementally/

Do you see what actually is causing the problem on Windows?

**Pull request merged**